### PR TITLE
refactor sampling tests to be context independent

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The sampling tests previously depended on the implementation of `Context`. This patch removes the dependence by depending the resulting traces rather than the intermediate state of spans and context. A snapshot test for a variety of sampling scenarios is also added.

This will unblock #1936

### CODE REVIEW

File: tests/snapshots/tests.integration.test_integration.test_sampling.snap [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+   "service" nil
+   "resource" "trace2"
+   "error" 0
+   "span_id" 0
+   "trace_id" 1
+   "parent_id" nil
+   "start" 1611739614070713000
+   "duration" 63000
+   "meta" {"runtime-id" "88c29c5e9e1445a8bcce78bd1238a4d4"}
+   "metrics" {"_dd.limit_psr" 1.0
```
Comment (Kyle-Verhoog) [inline tests/snapshots/tests.integration.test_integration.test_sampling.snap:33]: @brettlangdon does this tag have to be on every root span? I don't see it on every root span.

Comment (brettlangdon) [reply tests/snapshots/tests.integration.test_integration.test_sampling.snap:33]: which root spans are you talking about? trace root, chunk root, service root?

Comment (Kyle-Verhoog) [reply tests/snapshots/tests.integration.test_integration.test_sampling.snap:33]: trace root

**Review by Yun-Kim — APPROVED** [review]
Looks good to me!

**Pull request merged**